src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -2689,15 +2689,18 @@ class RideConstructionWindow final : public Window
 
         const auto& ted = GetTrackElementDescriptor(trackType);
         const auto* trackBlock = ted.Block;
+        const auto* rideEntry = currentRide->GetRideEntry();
+        auto clearanceHeight = (rideEntry != nullptr) ? rideEntry->Clearance
```

Comment:
The clearance height is set here, based on what the object specifies. So there is a direct relation, and it’s fully under the control of the object.

Reply:
My bad I was slightly mistaken, it does indeed use clearance as the visual bounding boxes currently, this is not for everything the same however. The clearance height and visual bounding boxes should be separated, the bounding boxes are relevant to render them in correct order so this may not necessarily align with the amount of space the object occupies on the tile, if you want to have more space above an object you want the clearance higher than the bounding box of the sprite, the bounding box should be only as big as the actual sprite to avoid odd drawing.